test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -0,0 +1,538 @@
+    }
+    
+    It 'Null is allowed when Allownull attribute is set' {
+
+        function get-foo
+        {
+            param( [Parameter(mandatory=$true)][allownull()] $a )
+            $a
+        }
+
+        (get-foo -a $null) | Should Be $null
+
+    }
+
+    It 'Empty string is not allowed AllowEmptyString Attribute is not set' {
+       
+        function get-foo
+        {
+            param( [Parameter(mandatory=$true)][string] $a )
+            $a
+        }
+
+        { get-foo -a '' } | ShouldBeErrorID 'ParameterArgumentValidationErrorEmptyStringNotAllowed,get-foo'
+    }
+
+    It 'Empty string is allowed when AllowEmptyString Attribute is set' {
+        function get-foo
+        {
+            param( [Parameter(mandatory=$true)][allowemptystring()][string] $a )
+            $a
+        }
+
+        get-foo -a '' | Should Be ''
+    }
+
+    It 'Error should be reported when AllowEmptyCollection it not set' {        
+
+        function get-foo
+        {
+            param( [Parameter(mandatory=$true)][string[]] $a )
+            $a
+        }
+
+        { get-foo -a @() } | ShouldBeErrorId 'ParameterArgumentValidationErrorEmptyArrayNotAllowed,get-foo'
+    }
+
+    It 'Empty collection is allowed when allowEmptyCollection is set' {
+        function get-foo
+        {
+            param( [Parameter(mandatory=$true)][allowemptycollection()][string[]] $a )
+            $a
+        }
+
+        get-foo -a @() | Should Be $null
+    }
+
+    It 'Unspecified non-mandatory bool shouldn not cause exception' {
+        function get-foo
+        {
+            param([Parameter(Mandatory=$true, Position=0, ValueFromPipeline=$true)] $a,
+                  [System.Boolean] $b)
+            $a
+        }
+
+        42 | get-foo | Should be 42
+    }
+
+    It 'Parameter binding failure on Parameter1 should not cause parameter binding failure on Length' {
+        function get-foo
+        {
+          param(
+                [Parameter(ValueFromPipeline = $true)]
+                [int]
+                $Parameter1 = 10,
+        
+                [Parameter(ValueFromPipelineByPropertyName = $true)]
+                [int]
+                $Length = 100
+          )
+  
+          process  { 
+            $Length
+          }
+        }
+
+        'abc' | get-foo | Should Be 3
+    }
+
+    It 'Binding array of string to array of bool should fail (cmdletbinding)' {
+        function get-foo
+        {
+           [cmdletbinding()]
+           param ([bool[]] $Parameter ) $Parameter
+        }
+        { get-foo 'a','b' } | ShouldBeErrorId 'ParameterArgumentTransformationError,get-foo'
+    }
+
+    It "Binding array of string to array of bool should succeed" {
+        function get-foo
+        {
+           param ([bool[]] $Parameter ) $Parameter
+        }        
+        
+        $x = get-foo 'a','b'
+        $x[0] | Should be $true
+        $x[1] | Should be $true
+
+    }
+
+    Context 'default values not converted properly' {
```

Comment:
fixed in next iteration

---
In reply to: [79214652](https://github.com/Powershell/Powershell/pull/2218#discussion_r79214652) [](ancestors = 79214652)